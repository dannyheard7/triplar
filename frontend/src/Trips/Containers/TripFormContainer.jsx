import React from "react";
import TripForm from "Trips/Components/TripForm";

export default class TripFormContainer extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            trip: this.props.trip,
            errors: []
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }


    handleChange(event) {
        let trip = this.state.trip;
        trip[event.target.name] = event.target.value;

        this.setState({trip: trip});
    }

    handleSubmit(e) {
        e.preventDefault();

        this.props.apiFunction(this.state.trip).then(response => {
            if(response.status === 400) {
                this.setState({errors: response.data})
            } else {
                this.setState({trip: {}, errors: []});
                this.props.onSuccess(response.data);
            }
        });
    }

    render () {
        return (
            <TripForm onChange={this.handleChange} onSubmit={this.handleSubmit} submitLabel={this.props.submitLabel}
                      errors={this.state.errors} suggestions={this.state.suggestions} trip={this.state.trip} />
        )
    }

}

TripFormContainer.defaultProps = {
    trip: {}
};