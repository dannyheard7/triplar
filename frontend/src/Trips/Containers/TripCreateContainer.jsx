import React from "react";
import api from "utils/api.js"

import TripFormContainer from "./TripFormContainer.jsx";

export default class TripCreateContainer extends React.Component {
    constructor(props) {
        super(props);

        this.onSuccess = this.onSuccess.bind(this);
    }

    onSuccess(trip) {
        this.props.onTripCreate(trip);
    };

    render () {
        return (
            <div className="card">
                <TripFormContainer submitLabel="Create" onSuccess={this.onSuccess} apiFunction={api.createTrip} />
            </div>
        )
    }
}