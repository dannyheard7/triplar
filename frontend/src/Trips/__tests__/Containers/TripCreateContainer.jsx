import React from "react";
import {shallow} from "enzyme";

import ShallowRenderer from 'react-test-renderer/shallow';
const renderer = new ShallowRenderer();

import TripCreateContainer from "Trips/Containers/TripCreateContainer";
import TripFormContainer from "Trips/Containers/TripFormContainer";

describe('<TripCreateContainer />', () => {
    const event = {
        preventDefault: () => {},
        target: {
            value: "",
            name: "",
        },
    };

    test('renders correctly', () => {
        const result = renderer.render(<TripCreateContainer />);
        expect(result).toMatchSnapshot();
    });

    test('calls onSuccess after receiving success from <TripFormContainer />', (function () {
        const stub = jest.spyOn(TripCreateContainer.prototype, 'onSuccess');

        const container = shallow(<TripCreateContainer onTripCreate={jest.fn()}/>);
        container.find(TripFormContainer).prop('onSuccess')(event);

        expect(stub).toBeCalled();
    }));

});