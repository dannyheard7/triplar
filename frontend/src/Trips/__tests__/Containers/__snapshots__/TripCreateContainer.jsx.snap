// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TripCreateContainer /> renders correctly 1`] = `
<div
  className="card"
>
  <TripFormContainer
    apiFunction={[Function]}
    onSuccess={[Function]}
    submitLabel="Create"
    trip={Object {}}
  />
</div>
`;
